TextToSpeech/TextToSpeech.cpp

```diff
@@ -140,43 +100,43 @@ namespace WPEFramework
                     m_connectionCallback = NULL;
                 }
 
-    	    	if(m_sessionCallback)
-    	    	{				
-                    delete m_sessionCallback;
-                    m_sessionCallback = NULL;
-    	        }		
             }
         }
 
-        void TextToSpeech::restoreTextToSpeech()
-        {
-            if(m_ttsEnabled && m_ttsManager)
-            {
-                m_ttsManager->enableTTS(m_ttsEnabled);
-                LOGWARN("Restored TTS connection and settings");
-    	    }
+        uint32_t nextSpeechId() {
+            static uint32_t counter = 0;
+
+            if(counter >= 0xFFFFFFFF)
+                counter = 0;
+
+            return ++counter;
         }
 
-        uint32_t TextToSpeech::enableTTS(const JsonObject& parameters, JsonObject& response)
+        uint32_t TextToSpeech::enabletts(const JsonObject& parameters, JsonObject& response)
```

Comment:
Since as Eugene mentioned the apis can be called from any of the worker threads, should we use some kind of sync mechanism on all the top level APIs?

Reply:
Added mutex guard on every interface apis.